packages/engine/Source/Scene/CameraEventAggregator.js

```diff
@@ -219,20 +218,73 @@ function listenMouseButtonDownUp(aggregator, modifier, type) {
 
   aggregator._eventHandler.setInputAction(
     function () {
-      aggregator._buttonsDown = Math.max(aggregator._buttonsDown - 1, 0);
-      isDown[key] = false;
-      releaseTime[key] = new Date();
+      cancelMouseDownAction(getKey(type, undefined), aggregator);
+      for (const modifierName in KeyboardEventModifier) {
+        if (KeyboardEventModifier.hasOwnProperty(modifierName)) {
+          const modifier = KeyboardEventModifier[modifierName];
+          if (defined(modifier)) {
```

Comment:
Then, by extension, this should never be undefined and I believe this check is not needed either.

Looking at it further if you want to loop over the keys and values of this enum you could use [`Object.entries()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) like this:
```js
for (const [modifierName, modifier] of Object.entries(KeyboardEventModifier)) { ... }
```

But at that point it seems you actually just want all the values which you can get with [`Object.values()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values) and end up with code like:

```js
for (const modifier of Object.values(KeyboardEventModifier)) {
  const cancelKey = getKey(type, modifier);
  cancelMouseDownAction(cancelKey, aggregator);
}
```